.github/workflows/sg_tests.yml

```diff
@@ -0,0 +1,35 @@
+name: Sync gateway tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  downloadAndRunSG:
+    name: download-and-run-sg
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with: 
+          submodules: recursive
+      - name: Download-and-run-SG
+        run: |
+          curl -o sg.zip https://packages.couchbase.com/releases/couchbase-sync-gateway/3.0.0-beta02/couchbase-sync-gateway-community_3.0.0-beta02_x86_64.zip
+          tar -zxvf sg.zip
+          ./couchbase-sync-gateway/bin/sync_gateway vendor/couchbase-lite-core/Replicator/tests/data/config.json & sleep 10
```

Comment:
We can run two SG for testing both SSL and non-SSL. There is also a config for SSL in that directory.

Reply:
Also we should test using SSL instead of non-SSL where possible.